R/setup.R

```diff
@@ -0,0 +1,137 @@
+#' Setup a project library, cache, options
+#' 
+#' This can be placed as the first line of any/all scripts and it will
+#' be create a reproducible, self-contained project with R packages. 
+#' Some of these have direct relationships with \code{RequireOptions}
+#' and arguments in \code{setLibPaths} and \code{Require}.
+#' @param RPackageFolders One or more folders where R packages are 
+#'   installed to and loaded from. In the case of more than one 
+#'   folder provided, installation will only happen in the first one.
+#' @param RPackageCache See \code{?RequireOptions}.
+#' @param buildBinaries See \code{?RequireOptions}.
+#' @inheritParams setLibPaths
+#' @export
+#' @rdname setup
+#' 
+#' @examples 
+#' # Place this as the first line of a project
+#' \dontrun{
+#' Require::setup()
+#' 
+#' # To turn it off and return to normal
+#' Require::setupOff()
+#' }
+#'   
+setup <- function(RPackageFolders = getOption("RPackageFolders", "R"), 
+                  RPackageCache = getOption("RPackageCache", "~/.cache"), 
```

Comment:
A subdirectory of `~/.cache` should be used, perhaps `~/.cache/R/Require`

Reply:
`~/.cache/R/RequirePkgCache`?

Reply:
I think that makes it redundant (we know it's a cache directory, that's why it's in `.cache`. I'm just looking and comparing against other R packages:

E.g., `~/.cache/R/R.cache` and `~/.cache/R-crancache` are both present  on my system, with the latter keeping CRAN package sources in `~/.cache/R-crancache/cran/src/contrib/`.

Have you looked at `crancache`? Too bad it's not on CRAN :disappointed:

Reply:
I like that `crancache` is keeping cran, bioc, and 'other' packages in separate subdirectories. Would that be useful here? Binaries can also be stashed in an appropriate `bin` subdirectory (versioned to R e.g., `3.6` and `4.0`).

Reply:
Turns out, I can set the `Require.RPackageCache` to use the one that `crancache` uses... if that exists.

Reply:
we would need to be careful so as not to have them collide. any index `crancache` maintains would break if `Require` also modifying files there

Reply:
unelss we use crancache, we should keep them separate

Reply:
Boo. Yes, looks like they won't let anything else write there.

Reply:
Actually, it seems to work. It was a Windows snafoo

Reply:
Basically, I am not adding any dependency on `crancache`... it will simply look for `crancache` in your packages, and if it there, it will point this out to you, and you can choose to set (manually) the cache folder of binaries for `Require` to be the `crancache`  binary folder, with a warning.

Reply:
i agree that crancache can't be used (it's not on cran), so we should keep them separate. Require shouldn't use crancache directories at all, as this is likely to cause problems with both packages.